### PR TITLE
new traitor item buster arm

### PR DESCRIPTION
**Pull request by LazennG**
# Document the changes in your pull request

Disclaimer: All damage to living things is brute and done with 35 AP on the user's targeted limb

age old question of 'what if seismic arm was evil' finally answered.
   The buster arm is a prosthetic limb that can be fitted to either the left or right arm without surgery. Attaching the arm remove's the user's ability to wear gloves and gives the user 5 abilities:

   Wire snatch- 5 second cd- puts a grappling hook in the non-buster hand that lets the user either pull unanchored targets to them or pull themselves towards anchored targets. Living targets will be immobilized on hit for 1.5 seconds and items hit will be put into the user's open hand. The hook has a range of 8 and 3 charges before disappearing but can be put away before they're used up. If there's a solid object in front of the user, the target will instead be moved to the user's location and take 15 damage.
   
dreamseeker_KdriNTVUxn

  Grapple - 3 second cd- Prepare the hand for grabbing something and put the readied ability in your hand. On hitting an object, the user will lift it above their head and throw it forward 1.5 seconds later. If the object hits something solid, it'll inflict 50 damage to both itself and the other object and snowball. If it hits a person, the object will still take 50 damage and the person will take 7 damage and be immobilized for 1.5 seconds and get snowballed. The snowball stops after 5 spaces or when it hits a dense object that doesn't break or a wall, where the victim will get knocked down and immobilized for 1.5 seconds. If a person is thrown they take 15 damage, and if they're thrown towards a disposals chute they get flushed. For reference it would take 6 throws into a wall and nothing else to make a welding fuel tank explode on impact. If the snowball gets into space then it flies like it was thrown normally. If the user is targeting a limb that's at its limit for brute damage, they'll tear it off when throwing the victim. If the victim's torso has 50 damage and they have a tail, the user will tear that off when targeting the torso. 
  Vending machines can be picked up and thrown as well, being moveable afterwards by anybody else. Additionally mechs can be picked up and thrown but will stop flying on hitting a solid structure and disable the arm for 5 seconds.
  
dreamseeker_Z1rYGNzCW8

  Slam-0.7 second cd- The user takes whoever's in front of them and slams them onto the space behind them, doing 25 damage. If there's a solid object or person on that space, the victim and second person take 10 damage and the object takes 400. If there's a non-rwall on that space then the user instead takes 30 damage and the wall breaks. If there's a disposals chute on that space then the victim gets flushed and if there's a space tile there the victim gets thrown behind the user.

dreamseeker_3xinnpDxiN

  Mop the floor- 4 sec cd- Dash forward 5 spaces and stop after the distance is reached or the user runs into a solid object. For every tile a victim travels with the user, they take 8 damage. If the victim is rammed into a wall, they take a bonus 10 damage and are immobilized & knocked down for 1.5 and 1 seconds respectively. Dragging them into a solid object does 200 damage to the object and knocks down/immobilizes for 1. Other people in the path of the attack get swept up in it too. If the move spills out into space then the victim gets thrown ahead of the user.
  
dreamseeker_7jtsjnB8YA
 
  Megabuster-6 second cd technically- Loudly charge up the arm for 2 seconds. If there's nothing in your hands to cancel the move, you'll get an item in the buster arm's hand called a megabuster for 5 seconds. In those 5 seconds, it can be used to deal 400 damage to targeted anchored objects, crush non-brain items, and smash walls. The timer will only reach its end sooner than 5 seconds if the user breaks an rwall, hits a mech, or hits a person. Once the megabuster expires, the arm gets disabled and renders itself and all the moves unusable for 15 seconds. If a mech is hit, it takes 150 damage. If an unanchored object is hit, it takes 50 damage and is sent flying, taking 110 damage for every solid object it collides with. If a person is hit, they take 30 damage and also go flying. Both kinds of targets will cause a snowball effect for other unanchored things in their path until they hit an rwall or something that stays solid after being collided with for 400 damage like any piece of anchored machinery or a wall girder. Hitting a solid object or wall will deal 10 damage or 20 damage to living things in the snowball respectively, whose full flying distance is 7 spaces. Living things hit by the snowball will take 20 damage before being added to it. If the targeted limb is at its limit for brute damage then it will instead fall off, stunning the victim for 3 seconds.

dreamseeker_Ng73aZqItE

The arm is priced at 15 tc. I would have gone higher but my consultants keep reminding me of things I forget like how the arm isn't as permanent as a martial art and how unlikely it is to land half the moves without setup, not to mention the emp vulnerability and how a considerable fight can still be put up against it if you keep your distance. Also despite its destructive capacity i chose to not limit to hijack or martyr or whatever because unlike shit like his grace or romerol or bombs, this can be controlled.
When the limb itself is used inhand it swaps out whichever arm it'll be fitted to when applied to someone.

also forgot but all these moves still gib dead animals idk if i said that
# Spriting
image

# Why is it good for the game
 I tire of these backstab knives and storage implants and greenshifts. I think an item that's strong right out the gate that's hopefully cool enough will help push things along especially when the user encounters the common problem of the target being in a busy area and not having the means to beat back the bystanders when they inevitably step in.

# Wiki Documentation

buster_arm_left
buster_arm_right
https://wiki.yogstation.net/wiki/Syndicate_Items#Conspicuous_and_Dangerous_Weapons
 Everything above is heavy word vomit so i think summarizing it up to something along the lines of 'a mechanical prosthetic that can be attached on the spot. Gives the user abilities that can cause heavy damage to the surrounding area and a grappling hook used to reposition themselves and others. The abilities are all melee-focused however, so the risk and reward run high. Additionally, gloves will no longer fit on the user so the arm can't be easily hidden.'

# Changelog

:cl:  @thegoldencat413 @Chubbygummibear @cuackles @cowbot92 @Mqiib @slicerv and saucecat
rscadd: Adds the buster arm 
imageadd: added buster arm sprites
/:cl:
huge thanks to goldencat for spending actual hours at a time helping me balance this shit so it's usable in combat unlike the seismic arm, as well as chubby for coding help and helping to test the arm with everyone named. BIG thanks to cuackles too for the icon sprite of the arm.

### CODE REVIEW

**Comment by UltraBigRat** [comment]
nice to see some more loud traitor items.

**Comment by MajManatee** [comment]
The one thing I have been waiting for

**Review by KillerOrcaCora — CHANGES_REQUESTED** [review]
Some icons are teetering on the edge of their UI but whatever, though the mop the floor icon is a bit too rough for an icon.
mop

https://user-images.githubusercontent.com/82983051/213015408-8ebc6e99-55b4-481c-8f82-347b294012c3.mp4

Also, think the chain whip ability can get its own in-hand-sprite instead of reusing the captain's chain-of-command more reflecting its OBJ which looks sick as hell and maybe a get over here sound file?

Last thing a lot of the abilities seem to leave people's sprites upside down a lot more than normal and not reverting to the original positioning so maybe that can be fixed?  

flipped

Oh and forgot, The top red on the arm needs a tiny bit of a tone down its kind of a bit too neon. 

Capture

**Comment by LazennG** [comment]
> Reviewing this overall this is pretty good stuff, only a few things that I can see getting more attention. The first thing that stood out most to me was the slam ability, in hindsight its a better mop of the floor but in conjunction, it seems a bit much, I suggest giving the slam ability a longer cooldown like mop because this seems a bit too much chip on the dip for just one ability.

yeah that was the initial impression (i think? no clue what chip in the dip means) of people testing too until it was used against a sentient person and they couldn't be hit with it if they weren't cornered or immobilized with another move. Additionally, the cooldown is low so it could help fill the role that would be taken by a proper melee weapon, except this is gaudier to use since like everything in the moveset is very dependent on positioning.

>Some icons are teetering on the edge of their UI but whatever, though the mop the floor icon is a bit too rough for an icon.

fuck

>Also, think the chain whip ability can get its own in-hand-sprite instead of reusing the captain's chain-of-command more reflecting its OBJ which looks sick as hell and maybe a get over here sound file?

thanks but the original item that used that icon sprite also came with the chain of command sprite too. might give it a shot if i can get through the rest of the changes

>Last thing a lot of the abilities seem to leave people's sprites upside down a lot more than normal and not reverting to the original positioning so maybe that can be fixed?

tried doing this several times throughout the development of the seismic arm and buster arm but i'll give it another shot now that most of the other code is handled

**Comment by KillerOrcaCora** [comment]
image
Yall ass knew that move was off the hook

**Comment by KillerOrcaCora** [comment]
image

**Comment by cuackles** [comment]
ok so being that orca is unable to review this i will, 2 of the changes requested have been done, will there be any changes to the UI buttons ? @LazennG

**Comment by LazennG** [comment]
> ok so being that orca is unable to review this i will, 2 of the changes requested have been done, will there be any changes to the UI buttons ? @LazennG

Unless it's something you want done I don't have it planned, mainly because goldencat's schedule has been busier lately and i'm not sure when he'd find time to change it @cuackles.

**Review by cuackles — APPROVED** [review]
allright fair enogh

**Comment by JamieD1** [comment]
Seems a bit op, will review with people

File: code/game/objects/items/busterarm.dm [hunk 1]
```diff
@@ -0,0 +1,913 @@
...
+	..()
+	
+/obj/item/projectile/wire/proc/zip(mob/living/user, turf/open/target)
+	to_chat(user, span_warning("You pull yourself towards [target]."))
+	playsound(user, 'sound/magic/tail_swing.ogg', 10, TRUE)
+	user.Immobilize(0.2 SECONDS)//so it's not cut short by walking
+	user.throw_at(get_step_towards(target,user), 8, 4)
+
+/obj/item/projectile/wire/on_hit(atom/target)
+	var/mob/living/carbon/human/H = firer
```
Comment (Moltijoe) [inline code/game/objects/items/busterarm.dm:115]: should probably add a check to make sure H exists

**Comment by Moltijoe** [comment]
gah, why is testing other people's PRs so annoying!?

**Comment by Moltijoe** [comment]
I DON'T UNDERSTAND, i can't even see if anything has changed in the json, why does it JUST KEEP WANTING ME TO DO THIS COMMIT.

**Comment by Moltijoe** [comment]
fuck it, i'll just review via the website alone

File: code/game/objects/items/busterarm.dm [hunk 2]
```diff
@@ -0,0 +1,913 @@
...
+	desc = "Prepare your left hand for grabbing. Throw your target and inflict more damage if they hit a solid object. If the targeted limb is horribly bruised, you'll tear it off\
+	when throwing the victim."
+	hand_path = /obj/item/buster/graphand
+	school = "transmutation"
+	charge_max = 30
+	action_icon = 'icons/mob/actions/actions_arm.dmi'
+	action_icon_state = "lariat"
+	sound = 'sound/magic/fleshtostone.ogg'
+	clothes_req = FALSE
+	var/jumpdistance = 4
```
Comment (Moltijoe) [inline code/game/objects/items/busterarm.dm]: jumpdistance? am i missing where this gets used?

File: code/game/objects/items/busterarm.dm [hunk 3]
```diff
@@ -0,0 +1,913 @@
...
+
+/obj/effect/proc_holder/spell/targeted/buster/megabuster/cast(list/targets, mob/living/user)
+	var/obj/item/buster/megabuster/B = new()
+	user.visible_message(span_userdanger("[user]'s left arm begins crackling loudly!"))
+	playsound(user,'sound/effects/beepskyspinsabre.ogg', 60, 1)
+	if(do_after(user, 2 SECONDS, user, TRUE, stayStill = FALSE))
+		var/result = (user.put_in_l_hand(B))
+		if(!result)
+			to_chat(user, span_warning("You can't do this with your left hand full!"))
+		if(result)
```
Comment (Moltijoe) [inline code/game/objects/items/busterarm.dm]: just do an if/else using (user.put_in_l_hand(B)) as the condition

File: code/game/objects/items/busterarm.dm [hunk 4]
```diff
@@ -0,0 +1,913 @@
...
+	var/summon_path = /obj/item/gun/magic/wire
+
+/obj/effect/proc_holder/spell/targeted/buster/wire_snatch/cast(list/targets, mob/user)
+	for(var/obj/item/gun/magic/wire/T in user)
+		qdel(T)
+		to_chat(user, span_notice("The wire returns into your wrist."))
+		return
+	for(var/mob/living/carbon/C in targets)
+		var/GUN = new summon_path
+		C.put_in_r_hand(GUN)
```
Comment (Moltijoe) [inline code/game/objects/items/busterarm.dm:59]: if the arm can be on either side of the body, why put the wire specifically in the right hand?

File: code/game/objects/items/busterarm.dm [hunk 5]
```diff
@@ -0,0 +1,913 @@
...
+	clothes_req = FALSE
+	var/jumpdistance = 4
+
+/obj/effect/proc_holder/spell/targeted/touch/buster/grap/cast(list/targets, mob/user = usr)
+	if(!QDELETED(attached_hand))
+		remove_hand(TRUE)
+		to_chat(user, span_notice("[dropmessage]"))
+		return
+	for(var/mob/living/carbon/C in targets)
+		C.visible_message(span_warning("The fingers on [C]'s left buster arm begin to tense up."))
```
Comment (Moltijoe) [inline code/game/objects/items/busterarm.dm:188]: if the arm can be on either side of the body, why the left arm?

File: code/game/objects/items/busterarm.dm [hunk 6]
```diff
@@ -0,0 +1,913 @@
...
+						user.put_in_l_hand(limb_to_hit)
+					else
+						user.put_in_r_hand(limb_to_hit)
+			if(limb_to_hit == L.get_bodypart(BODY_ZONE_PRECISE_GROIN))
+				var/obj/item/organ/T = L.getorgan(/obj/item/organ/tail)
+				if(T && limb_to_hit.brute_dam >= 50)
+					to_chat(L, span_userdanger("[user] tears your tail off!"))
+					playsound(user,'sound/misc/desceration-02.ogg', 20, 1)
+					L.visible_message(span_warning("[user] throws [L], severing [L.p_them()] from [L.p_their()] tail!")) //"I'm taking this back."
+					T.Remove(L)
```
Comment (Moltijoe) [inline code/game/objects/items/busterarm.dm:295]: based

**Review by Moltijoe — CHANGES_REQUESTED** [review]
Seems to function exactly as stated from the videos, but these could maybe be problems
also something fucked up when trying to test the PR and am now stuck in a limbo of json commit

File: code/game/objects/items/busterarm.dm (hunk 5 above)
Comment (LazennG) [reply code/game/objects/items/busterarm.dm:188]: ah fuck missed that thanks

Comment (LazennG) [reply code/game/objects/items/busterarm.dm:188]: > if the arm can be on either side of the body, why the left arm?

wait no disregard what i said before this is cause this spell is specifically for the corresponding arm hence the other version existing

File: code/game/objects/items/busterarm.dm (hunk 4 above)
Comment (LazennG) [reply code/game/objects/items/busterarm.dm:59]: the default version of the spell is for the left hand while the subtype is for the right

File: code/game/objects/items/busterarm.dm (hunk 2 above)
Comment (LazennG) [reply code/game/objects/items/busterarm.dm]: whoops yeah this doesnt get used

**Review by Moltijoe — APPROVED** [review]
Can't find anything else wrong just looking at the code
i'm unable to test it because of my branch being mucked up, but the videos you have posted at least show it works.
All that needs to happen now is jamie approving it because of what i did when trying to test this.

**Comment by kooarbiter** [comment]
it's fucking perfect, one small change I'd add is doing the suplex with it should give your target a swirlie if they land on a toilet

File: code/game/objects/items/busterarm.dm [hunk 7]
```diff
@@ -0,0 +1,910 @@
...
+						playsound(L,'sound/effects/pop_expl.ogg', 20, 1)
+						return
+				L.forceMove(T)
+	if(iswallturf(target))
+		var/turf/W = target
+		zip(H, W)
+
+/obj/item/gun/magic/wire/process_chamber()
+	. = ..()
+	if(charges == 0)
```
Comment (Bibby0110) [inline code/game/objects/items/busterarm.dm]:
```suggestion
	if(!charges)
```

File: code/game/objects/items/busterarm.dm [hunk 8]
```diff
@@ -0,0 +1,910 @@
...
+		if(!attached_hand)
+			attached_hand = new hand_path(src)
+			attached_hand.attached_spell = src
+			user.put_in_r_hand(attached_hand)
+			recharging = FALSE
+			if(C.active_hand_index % 2 == 1)
+				C.swap_hand(0)
+			return
+
+/obj/effect/proc_holder/spell/targeted/buster/megabuster/right
```
Comment (Bibby0110) [inline code/game/objects/items/busterarm.dm]:
```suggestion
```

**Comment by UltraBigRat** [comment]
So i gave this a whirl recently and must say,  while it looks absolutely sick its pretty underpowered, the 15tc cost means you cant really get much else in good weapons or  defensive tools other then like uno cards while the arms damage is pretty low, is clunky to use with all the buttons  and difficult landings,  and it has no way to recover so its very simple to just disabler it and they cant do much, id reccomend either buffing it with some form of recovery, or  making its cost lower, cause right now its just not worth 15tc in my humble opinion, with which you could get many other far better options.

**Pull request merged**